05-services.md

```diff
@@ -1393,6 +1393,14 @@ If `pull_policy` and `build` are both present, Compose implementations SHOULD bu
 
 `read_only` configures service container to be created with a read-only filesystem.
 
+### remove
+
+When set to `true`, `remove` will configure platform to automatically remove the
+resource when the process exits. This has no effect if `restart` is set.
+
+Service continuity MAY be affected, as resource is removed before Compose implementation
+can react to process termination. Typically, anonymous volumes will be lost.
```

Comment:
> before Compose implementation can react to process termination

Should we disallow having `depends_on` reference a service that has `remove: true`? Or is there something else you're concerned about here?

(Sidenote: is it possible to define anonymous volumes in Compose?)

Reply:
The challenge is not about dependencies but about container restart & replacement. Inheriting anonymous volumes is the first to come in my mind but there's probably a few other corner-cases.
I can see benefit of this feature with container used with `docker compose run` to avoid orphaned. For services used with `up` this is not so obvious.

> Sidenote: is it possible to define anonymous volumes in Compose

nope, anonymous are only defined by image metadata (Dockerfile's `VOLUME`)  when no explicit binding is defined in container config.